src/Employer/Employer.Web/Controllers/Part2/TrainingProviderController.cs

```diff
@@ -41,7 +41,10 @@ public async Task<IActionResult> SelectTrainingProvider(SelectTrainingProviderEd
                 return View(vm);
             }
 
-            var providerExists = await _orchestrator.ConfirmProviderExists(long.Parse(m.Ukprn));
+            if (long.TryParse(m.Ukprn, out var ukprnAsLong) == false)
+                return await ProviderNotFound(m);
```

Comment:
This IF statement is just belt-and-braces. The regex validation would prevent us getting this far.